projects/core/src/cart/facade/cart.service.ts

```diff
@@ -223,4 +225,67 @@ export class CartService {
   private isLoggedIn(userId: string): boolean {
     return typeof userId !== 'undefined';
   }
+
+  // TODO: Remove once backend is updated
+  /**
+   * Temporary method to merge guest cart with user cart because of beackend limitation
+   * This is for an edge case
+   */
+  private guestCartMerge(): void {
+    let cartEntries: OrderEntry[];
+    this.getEntries()
+      .pipe(take(1))
+      .subscribe(entries => {
+        cartEntries = entries;
+      });
+
+    this.store.dispatch(
+      new CartActions.DeleteCart({
+        userId: ANONYMOUS_USERID,
+        cartId: this.cartData.cart.guid,
+      })
+    );
+
+    combineLatest([
```

Comment:
`this.store.pipe(select(CartSelectors.getActiveCartState)` can get `cart content `and `loading flag` in one select.